SignalMessaging/contacts/SystemContactsFetcher.swift

```diff
@@ -285,10 +285,24 @@ public class SystemContactsFetcher: NSObject {
     private func updateContacts(completion completionParam: ((Error?) -> Void)?, isUserRequested: Bool = false) {
         AssertIsOnMainThread()
 
+        var backgroundTask: OWSBackgroundTask? = OWSBackgroundTask(label: "\(#function)", completionBlock: { [weak self] status in
```

Comment:
Systems contact update has two phases (fetch, parse), both of which can be costly.

Arguably we should abort both phases when we enter the background.  But we'd probably want to try again when we re-entered the foreground.